constraint_generation/src/execution_data/executed_template.rs

```diff
@@ -82,6 +89,27 @@ impl ExecutedTemplate {
         self.intermediates.push((intermediate_name.to_string(), dimensions.to_vec()));
     }
 
+    pub fn add_ordered_signal(&mut self, signal_name: &str, dimensions: &[usize]) {
```

Comment:
When executing a signal declaration inside a template's body we want to store that signal in the order of declaration. But it's important to note that if the signal is an array, we should generate all symbols for all possible positions in that array. That is achieved with the generate_symbols function here.